js/privatebin.js

```diff
@@ -4149,65 +4485,197 @@ jQuery.PrivateBin = (function($, sjcl, Base64, RawDeflate) {
-                TopNav.setRetryCallback(function () {
-                    TopNav.hideRetryButton();
+            // shows the remaining time (until) deletion
+            PasteStatus.showRemainingTime(paste);
 
-                    me.run(paste);
+            Promise.all(decryptionPromises)
+                .then(() => {
+                    Alert.hideLoading();
+                    TopNav.showViewButtons();
+                })
+                .catch((err) => {
+                    // wait for the user to type in the password,
+                    // then PasteDecrypter.run will be called again
+                    console.error(err);
                 });
-                TopNav.showRetryButton();
-            }
         };
 
+        return me;
+    })();
+
+
+    /**
+     * initial (security) check
+     *
+     * @name   InitialCheck
+     * @param  {object} window
+     * @param  {object} document
+     * @class
+     */
+    var InitialCheck = (function () {
+        var me = {};
+
         /**
-         * initialize
+         * blacklist of UserAgents (parts) known to belong to a bot
          *
-         * @name   PasteDecrypter.init
+         * @private
+         * @enum   {Array}
+         * @readonly
+         */
+        const badBotUA = [
+            'Bot',
+            'bot'
+        ];
+
+        /**
+         * blacklist of UserAgent versions known not to work with this application
+         *
+         * @private
+         * @enum   {Object}
+         * @readonly
+         */
+        const oldUA = [
+            {
+                'regex': /Chrome\/([0-9]+)/,
+                'minVersion': 57,
+            },
+            {
+                'regex': /Edge\/([0-9]+)/,
+                'minVersion': 16,
+            },
+            {
+                'regex': /Firefox\/([0-9]+)/,
+                'minVersion': 54,
+            },
+            {
+                'regex': /Opera\/.*Version\/([0-9]+)/,
+                'minVersion': 44,
+            },
+            {
+                'regex': /Version\/([0-9]+).*Safari/,
+                'minVersion': 11,
+            }
+        ];
+
+        /**
+         * check if the connection is insecure
+         *
+         * @private
+         * @name   InitialCheck.isInsecureConnection
          * @function
+         * @return {bool}
+         */
+        function isInsecureConnection()
+        {
+            const url = new URL(document.URL);
+
+            // HTTP is obviously insecure
+            if (url.protocol !== 'http:') {
+                return false;
+            }
+
+            // filter out actually secure connections over HTTP
+            if (
+                url.hostname.endsWith('.onion') ||
+                url.hostname.endsWith('.i2p')
+            ) {
+                return false;
+            }
+
+            // whitelist localhost for development
+            if (
+                url.hostname === 'localhost' ||
+                url.hostname === '127.0.0.1'
```

Comment:
Considering `127.*.*.*` (subnet `127.0.0.0/8`) is actually all reserved for local domains, maybe we should whitelist it all?

Reply:
Yeah, let me see if we can move this into an array or so at the top of the object as well, like the bot and user agent stuff.

Reply:
Oh, and I use IPv6 extensively, so ::1 etc. can be added as well.

Reply:
yeah… (and is not it also a subnet for IPv6, dunno…?)